prepare_oim/roles/deploy_containers/common/tasks/configure_chrony.yml

```diff
@@ -31,10 +31,75 @@
     enabled: true
     state: restarted
 
-- name: Chronyc sources
-  ansible.builtin.command: chronyc sources
-  changed_when: false
-  register: chronyc_sources
-  until: chronyc_sources is not failed
-  retries: "{{ hostvars['localhost']['fail_retry'] }}"
-  delay: "{{ hostvars['localhost']['fail_delay'] }}"
+- name: Retrieve NTP servers
+  ansible.builtin.set_fact:
+    ntp_servers: "{{ hostvars['localhost']['ntp_servers'] | default([]) }}"
+
+- name: Configure NTP servers in chrony when ntp_servers are provided
+  block:
+    - name: Check reachability of configured NTP servers
+      ansible.builtin.command: "ping -c 1 {{ item.address }}"
+      register: ntp_ping_results
+      changed_when: false
+      failed_when: false
+      loop: "{{ ntp_servers | default([]) }}"
+
+    - name: Build list of reachable NTP servers
+      ansible.builtin.set_fact:
+        reachable_ntp_servers: "{{ ntp_ping_results.results | selectattr('rc', 'equalto', 0) | map(attribute='item') | list }}"
+
+    - name: Remove existing NTP server and pool entries from chrony.conf
+      ansible.builtin.lineinfile:
+        path: "{{ chrony_conf_path }}"
+        regexp: '^(server|pool)\s+'
+        state: absent
+      when:
+        - reachable_ntp_servers is defined
+        - reachable_ntp_servers | length > 0
+
+    - name: Manage NTP server entries in chrony.conf
+      ansible.builtin.blockinfile:
+        path: "{{ chrony_conf_path }}"
+        marker: "# {mark} ANSIBLE MANAGED NTP SERVERS"
+        block: |
+          {% for srv in reachable_ntp_servers %}
+          {{ srv.type }} {{ srv.address }} iburst
+          {% endfor %}
+      when:
+        - reachable_ntp_servers is defined
+        - reachable_ntp_servers | length > 0
+
+    - name: Enable and start chronyd service
+      ansible.builtin.service:
+        name: chronyd
+        enabled: true
+        state: restarted
+
+    - name: Chronyc sources
```

Comment:
few tasks should be outside block as well. it has to be configured with ntp servers also